erpnext/regional/saudi_arabia/utils.py

```diff
@@ -143,5 +145,6 @@ def delete_vat_settings_for_company(doc, method):
 	if doc.country != 'Saudi Arabia':
 		return
 
-	settings_doc = frappe.get_doc('KSA VAT Setting', {'company': doc.name})
-	settings_doc.delete()
\ No newline at end of file
+	settings_doc = frappe.get_all('KSA VAT Setting', filters={'company': doc.name}, pluck='name')
```

Comment:
There is only one settings doc per company and this hook is called on deletion of the company, why do we need to use get_all?

Reply:
I believe the settings doc of the company can be manually deleted by the user. Therefore, we will have to check if the document exists by name and then delete it. How about restricting the delete permission?